pkg/apis/eksctl.io/v1alpha5/vpc.go

```diff
@@ -64,6 +64,48 @@ func (m *AZSubnetMapping) SetAZ(az string, spec Network) {
 	}
 }
 
+// WithIDs returns list of subnet ids
+func (m *AZSubnetMapping) WithIDs() []string {
+	if m == nil {
+		return nil
+	}
+	subnets := []string{}
+	for _, s := range *m {
+		if s.ID != "" {
+			subnets = append(subnets, s.ID)
+		}
+	}
+	return subnets
+}
+
+// WithCIDRs returns list of subnet CIDRs
+func (m *AZSubnetMapping) WithCIDRs() []string {
+	if m == nil {
+		return nil
+	}
+	subnets := []string{}
+	for _, s := range *m {
+		if s.CIDR != nil {
+			subnets = append(subnets, s.CIDR.String())
+		}
+	}
+	return subnets
```

Comment:
do we need to consider if a subnet has provided its ID and the cidr range?

Reply:
hmm I suppose not. that reminds me also I need to add validation

Reply:
going to look at that in a new PR though

Reply:
:+1: